File: ActionsModule/ActionsLib/Team/Player.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Numerics;
using System.Text;
using System.Threading.Tasks;

namespace ActionsLib
{
    public class Player
    {
        string _name;
        string _surname;
        int _height;
        int _number;
        Amplua _amplua;

        public Player(string name, string surname, int height, int number, Amplua amplua)
        {
            _name = name;
            _surname = surname;
            _height = height;
            _number = number;
            _amplua = amplua;
        }

        public int Number
        {
            get { return _number; }
        }
        public Amplua Amplua
        {
            get { return _amplua; }
            set { _amplua = value; }
        }
        public string Surname
        {
            get { return _surname; }
        }
        public int Height
        {
            get { return _height; }
        }
        public string Name
        {
            get { return _name; }
        }
        public static bool operator ==(Player left, Player right)
        {
            if (ReferenceEquals(left, null) && ReferenceEquals(right, null))
                return true;
            if (ReferenceEquals(left, null) || ReferenceEquals(right, null))
                return false;
            return left.Number == right.Number;
        }
        public static bool operator !=(Player left, Player right)
        {
            if (ReferenceEquals(left, null) && ReferenceEquals(right, null))
                return false;
            if (ReferenceEquals(left, null) || ReferenceEquals(right, null))
                return true;
            return left.Number != right.Number;
        }

        public string ToTableString()
        {
            return $"{_number} {_surname} {_name}";
        }
    }

    public enum Amplua
    {
        Libero = 0,
        Setter = 1,
        OutsideHitter = 2,
        MiddleBlocker = 3,
        Opposite = 4,
        Undefined = -1
    }
}

File: volleyball_stats/app/models/player.py
"""
Модель игрока
"""
from enum import IntEnum
from typing import Optional
from pydantic import BaseModel, Field


class Amplua(IntEnum):
    """Амплуа игрока"""
    LIBERO = 0
    SETTER = 1
    OUTSIDE_HITTER = 2
    MIDDLE_BLOCKER = 3
    OPPOSITE = 4
    UNDEFINED = -1


class Player(BaseModel):
    """Модель игрока волейбольной команды"""
    name: str = Field(..., description="Имя игрока")
    surname: str = Field(..., description="Фамилия игрока")
    height: int = Field(..., ge=0, description="Рост игрока в см")
    number: int = Field(..., ge=0, description="Номер игрока")
    amplua: Amplua = Field(default=Amplua.UNDEFINED, description="Амплуа игрока")

    class Config:
        use_enum_values = False
        json_schema_extra = {
            "example": {
                "name": "Иван",
                "surname": "Иванов",
                "height": 195,
                "number": 10,
                "amplua": 2
            }
        }

    def to_table_string(self) -> str:
        """Представление для таблицы"""
        return f"{self.number} {self.surname} {self.name}"

    def __str__(self) -> str:
        return f"#{self.number}"

    def __eq__(self, other) -> bool:
        if not isinstance(other, Player):
            return False
        return self.number == other.number

    def __hash__(self):
        return hash(self.number)
